### PR TITLE
Support worker server to run bat script

### PR DESCRIPTION
**Pull request by liwenhe1993**
Support worker server to run bat script

1. Reimplement ProcessImpl.java, ProcessEnvironment.java and ProcessBuilder.java for Windows
2. Modify shell task code for windows
3. Add ASF License

#2015

### CODE REVIEW

**Comment by liwenhe1993** [comment]
This PR has a problem with npm can't build it.

**Comment by samz406** [comment]
sonarcloud scanned some issues, you can deal with

**Comment by liwenhe1993** [comment]
For now, npm build failed, dont merge.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessBuilderForWin32.java [hunk 1]
```diff
@@ -0,0 +1,1065 @@
...
+ *
+ * <p>To start a process with an explicit set of environment
+ * variables, first call {@link Map#clear() Map.clear()}
+ * before adding environment variables.
+ *
+ * @author Martin Buchholz
+ * @since 1.5
+ */
+
+public class ProcessBuilderForWin32 {
```
Comment (TisonKun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessBuilderForWin32.java]: @dailidong what's the different between this class and the internal class? I would prefer remove all duplicated document, emphasize the difference and link to ProcessBuilder's build if there is nothing changed. Also at the beginning of the class document, it is possibly required that we mention it is copied from openjdk with modification, which will help reduce legal risk.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessEnvironmentForWin32.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dolphinscheduler.common.utils.process;
+
+import com.sun.jna.platform.win32.Kernel32Util;
+
+import java.util.*;
+
+final class ProcessEnvironmentForWin32 extends HashMap<String,String> {
```
Comment (TisonKun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessEnvironmentForWin32.java]: ditto

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessImplForWin32.java [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.sun.jna.platform.win32.WinBase.STILL_ACTIVE;
+
+public class ProcessImplForWin32 extends Process {
```
Comment (TisonKun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/process/ProcessImplForWin32.java]: ditto

**Pull request merged**